pkg/projectfile/projectfile.go

```diff
@@ -437,17 +439,40 @@ var persistentProject *Project
 
 // Parse the given filepath, which should be the full path to an activestate.yaml file
 func Parse(configFilepath string) (*Project, *failures.Failure) {
-	dat, err := ioutil.ReadFile(configFilepath)
+	projectDir := filepath.Dir(configFilepath)
+	files, err := ioutil.ReadDir(projectDir)
 	if err != nil {
-		return nil, failures.FailIO.Wrap(err)
+		return nil, failures.FailIO.Wrap(err, locale.Tl("err_project_readdir", "Could not check for project files in your project directory."))
 	}
 
-	project := Project{}
-	err = yaml.Unmarshal([]byte(dat), &project)
-	project.path = configFilepath
+	project, fail := parse(configFilepath)
+	if fail != nil {
+		return nil, fail
+	}
 
-	if err != nil {
-		return nil, FailParseProject.New(locale.T("err_project_parse", map[string]interface{}{"Error": err.Error()}))
+	re, _ := regexp.Compile(`activestate.(\w+).yaml`)
+	for _, file := range files {
```

Comment:
nit: fileInfo and fileInfos would better reflect what type is being handled here.